presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorSplitManager.java

```diff
@@ -23,7 +23,8 @@ ConnectorSplitSource getSplits(
             ConnectorTransactionHandle transactionHandle,
             ConnectorSession session,
             ConnectorTableLayoutHandle layout,
-            SplitSchedulingStrategy splitSchedulingStrategy);
+            SplitSchedulingStrategy splitSchedulingStrategy,
+            boolean schedulerUsesAddresses);
```

Comment:
I haven't looked into the context yet; but by only referring from the name of this flag, I probably won't be able to tell to set it to true or false is the correct call for my next new connector. Can we merge this field into `SplitSchedulingStrategy`? Or infer it from other way?

Reply:
This is only a hint from the engine. No matter whether it is true or false - you can either return, or don't return the split addresses. It is just if it is set to `false` it indicates that even if you provide split locations to the engine - they still won't be used. But you are still free to provide them.

Reply:
That's exactly the problem, isn't it? The interface will have something ambiguous. A javadoc would be appreciated. I would highly suggest merging this into `SplitSchedulingStrategy` (if possible). Make it with a default value `true` maybe.

Reply:
I'll create a class `SplitSchedulingContext` which will contain SplitSchedulingStrategy and a boolean `schedulerUsesAddresses`.  The two can't be combined to a single fieldbecasue splitschedulingstrategy is about whether it uses grouped execution and this is about whether the scheduling is topology aware (so how splits are scheduled vs. where they are scheduled). I'll add javadoc with a good explanation.